File: packages/react-native-test-app-msal/android/src/main/java/com/microsoft/reacttestapp/msal/AuthResult.kt
package com.microsoft.reacttestapp.msal

import com.microsoft.identity.client.IAuthenticationResult

data class AuthResult(
    val accessToken: String,
    val username: String,
    val expirationTime: Int,
    val redirectUri: String
) {
    constructor(result: IAuthenticationResult, redirectUri: String) : this(
        result.accessToken,
        result.account.username,
        (result.expiresOn.time / 1000).toInt(),
        redirectUri
    )
}

File: incubator/react-native-test-app-msal/ios/AuthResult.swift
import Foundation
import MSAL

@objcMembers
public final class AuthResult: NSObject {
    public let accessToken: String
    public let username: String
    public let expirationTime: Int
    public let redirectURI: String?

    public init(accessToken: String, username: String, expirationTime: Int, redirectURI: String? = nil) {
        self.accessToken = accessToken
        self.username = username
        self.expirationTime = expirationTime
        self.redirectURI = redirectURI
    }
}

extension AuthResult {
    convenience init(result: MSALResult) {
        self.init(
            accessToken: result.accessToken,
            username: result.account.username ?? "",
            expirationTime: Int(result.expiresOn?.timeIntervalSince1970 ?? 0),
            redirectURI: TokenBroker.Constants.RedirectURI
        )
    }
}
